internal/executor/podchecks/failedpodchecks/event_checks.go

```diff
@@ -0,0 +1,68 @@
+package failedpodchecks
+
+import (
+	"fmt"
+	"regexp"
+
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/armadaproject/armada/internal/executor/configuration/podchecks"
+)
+
+type eventRetryChecker interface {
+	IsRetryable(podEvents []*v1.Event) (bool, string)
+}
+
+type podEventCheck struct {
+	regexp    *regexp.Regexp
+	reason    string
+	eventType podchecks.EventType
+}
+
+type PodEventsChecker struct {
+	checks []podEventCheck
+}
+
+func NewPodEventsChecker(checks []podchecks.PodEventCheck) (*PodEventsChecker, error) {
+	podEventChecks := make([]podEventCheck, 0, len(checks))
+
+	for _, check := range checks {
+		re, err := regexp.Compile(check.Regexp)
```

Comment:
At some point it would be nice to see if we can get viper to marshall regexes and push stuff like this to config validation.